### PR TITLE
Bugfix: fix random val/test split issue

### PR DESCRIPTION
**Pull request by djdameln**
This PR solves the problems with incorrect train/val/test splits. The seed from the config is now used when the splits are created. Also the `create_validation_set` argument of the `make_mvtec_dataset` now takes effect when set to True.

### CODE REVIEW

**Review by ashwinvaidya17 — APPROVED** [review]
This was a good catch!

File: anomalib/data/mvtec.py [hunk 1]
```diff
@@ -199,10 +199,10 @@ def make_mvtec_dataset(
     samples.label_index = samples.label_index.astype(int)
 
     if create_validation_set:
-        samples = create_validation_set_from_test_set(samples)
+        samples = create_validation_set_from_test_set(samples, seed=seed)
 
     # Get the data frame for the split.
-    if split is not None and split in ["train", "test"]:
+    if split is not None and split in ["train", "test", "val"]:
```
Comment (samet-akcay) [inline anomalib/data/mvtec.py:205]: `'train', 'val' or 'test'` as the logical flow

File: anomalib/data/mvtec.py [hunk 2]
```diff
@@ -415,47 +424,45 @@ def __init__(
...
-            root=self.root,
-            category=self.category,
-            pre_process=self.pre_process,
-            is_train=False,
-            download=True,
-        )
+        self.train_data: Dataset
+        self.test_data: Dataset
+        if create_validation_set:
+            self.val_data: Dataset
```
Comment (samet-akcay) [inline anomalib/data/mvtec.py:433]: Is flake and pylint ok with this? When we don't define the variables in `__init__`, they usually complain.

File: anomalib/data/mvtec.py [hunk 3]
```diff
@@ -464,8 +471,12 @@ def train_dataloader(self) -> DataLoader:
 
     def val_dataloader(self) -> DataLoader:
         """Get validation dataloader."""
-        return DataLoader(self.val_data, shuffle=False, batch_size=self.test_batch_size, num_workers=self.num_workers)
+        if self.create_validation_set:
+            return DataLoader(
+                self.val_data, shuffle=False, batch_size=self.test_batch_size, num_workers=self.num_workers
+            )
+        return DataLoader(self.test_data, shuffle=False, batch_size=self.test_batch_size, num_workers=self.num_workers)
```
Comment (samet-akcay) [inline anomalib/data/mvtec.py:478]: Maybe we could simplify this as follows:

```
dataset = self.val_data if self.create_validation_set else self.test_data
return DataLoader(dataset=dataset, shuffle=False, batch_size=self.test_batch_size, num_workers=self.num_workers)
```

**Comment by samet-akcay** [comment]
Thanks @djdameln! Nice one! Only minor comments.

**Review by samet-akcay — APPROVED** [review]
